src/MarginTrading.Backend.Services/VolumeEquivalentService.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using MarginTrading.Backend.Core;
+using MarginTrading.Backend.Core.Settings;
+
+namespace MarginTrading.Backend.Services
+{
+	public class VolumeEquivalentService : IVolumeEquivalentService
+	{
+		private readonly ICfdCalculatorService _cfdCalculatorService;
+		private readonly string _equivalentAssetSetting;
+
+		public VolumeEquivalentService(
+			ICfdCalculatorService cfdCalculatorService,
+			string equivalentAssetSetting)
+		{
+			_cfdCalculatorService = cfdCalculatorService;
+			_equivalentAssetSetting = equivalentAssetSetting;
+		}
+
+		private string GetEquivalentAsset()
+		{
+			return _equivalentAssetSetting;
+		}
+
+		public void EnrichOpeningOrder(Order order)
+		{
+			order.EquivalentAsset = GetEquivalentAsset();
+			order.OpenPriceEquivalent = _cfdCalculatorService.GetQuoteRateForQuoteAsset(order.EquivalentAsset,
+				order.Instrument);
+		}
+
+		public void EnrichClosingOrder(Order order)
+		{
+			if (string.IsNullOrEmpty(order.EquivalentAsset))
+				order.EquivalentAsset = GetEquivalentAsset();
```

Comment:
Please add at least a blank line after an `if` without curved bracers, otherwise it can be dangerous and unreadable.

Reply:
Ok, but it's a point of design.